colossus/src/main/scala/colossus/protocols/http/HttpResponse.scala

```diff
@@ -116,9 +115,12 @@ object HttpResponse extends HttpResponseBuilding {
   def initialVersion = HttpVersion.`1.1`
 
   def apply[T: HttpBodyEncoder](head: HttpResponseHead, body: T): HttpResponse = {
-    HttpResponse(head, HttpBody(implicitly[HttpBodyEncoder[T]].encode(body)))
+    val encoder = implicitly[HttpBodyEncoder[T]]
+    val response = HttpResponse(head, HttpBody(encoder.encode(body)))
+    encoder.contentType.fold(response) { ct => response.withContentType(ct) }
   }
 
+  // THESE ARE ONLY USED IN TESTS!
```

Comment:
Not true, remember this is a framework, so just because a method isn't used by the framework itself doesn't mean a user of the framework isn't using it :)